pdr_backend/lake/partitioned_data_store.py

```diff
@@ -0,0 +1,114 @@
+import hashlib
+import os
+import polars as pl
+import duckdb
+from enforce_typing import enforce_types
+
+
+class PartitionedDataStore:
+    @enforce_types
+    def __init__(self, base_directory=str):
+        self.base_directory = base_directory
+        self.duckdb_conn = duckdb.connect(
+            database=":memory:"
+        )  # Keep a persistent connection
+
+    @enforce_types
+    def _generate_view_name(self, base_path=str) -> str:
+        """
+        Generate a unique view name for a given base path.
+        @arguments:
+            base_path - The base path to generate a view name for.
+        @returns:
+            str - A unique view name.
+        """
+
+        hash_object = hashlib.md5(base_path.encode())
+        return f"dataset_{hash_object.hexdigest()}"
+
+    @enforce_types
+    def append_data(self, df: pl.DataFrame, dataset_identifier: str):
+        """
+        Append data to an in-memory dataset.
+        @arguments:
+            df - The Polars DataFrame to append.
+            dataset_identifier - A unique identifier for the dataset.
+        """
+
+        view_name = self._generate_view_name(self.base_directory + dataset_identifier)
+        # Register the DataFrame as an Arrow table temporarily
+        # arrow_table = df.to_arrow()
+        self.duckdb_conn.register(view_name, df)
```

Comment:
Although this is called append, it doesn't technically append.
As far as I can tell, this re-creates a new view every time and registers the table.

But, if the table already exists, then shouldn't this insert rows into the duckdb table?